server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -101,6 +104,7 @@ public IngestDocument(String index, String id, long version, String routing, Ver
      */
     public IngestDocument(IngestDocument other) {
         this(
+            other.originalMetadata.clone(),
```

Comment:
I've thought about this for a little bit and I don't think it's a requirement. This clone operation is applicable to two metadata objects: The original one that should never receive an update, and the mutable version of it which sees updates whenever a document's index or id or some other metadata changes. It might be a smidge better in performance but I think I'd want to make double sure.